frost-signer/src/signing_round.rs

```diff
@@ -557,13 +628,47 @@ impl SigningRound {
     }
 
     fn dkg_private_begin(&mut self) -> Result<Vec<MessageTypes>, Error> {
+        let mut rng = OsRng::default();
         let mut msgs = vec![];
         for (key_id, private_shares) in &self.signer.frost_signer.get_shares() {
-            info!("sending dkg private share for key_id #{}", key_id);
+            info!(
+                "signer {} sending dkg private share for key_id #{}",
+                self.signer.signer_id, key_id
+            );
+            // encrypt each share for the recipient
+            let mut encrypted_shares = HashMap::new();
+
+            for (dst_key_id, private_share) in private_shares {
+                debug!(
+                    "encrypting dkg private share for key_id #{}",
+                    dst_key_id + 1
+                );
+                let compressed =
+                    Compressed::from(self.signer_keys.key_ids[&(dst_key_id + 1)].to_bytes());
+                let dst_public_key = Point::try_from(&compressed).unwrap();
+                let shared_key = self.network_private_key * dst_public_key;
+                let mut nonce_bytes = [0u8; 12];
+
+                rng.fill_bytes(&mut nonce_bytes);
+
+                let nonce_vec = nonce_bytes.to_vec();
+                let nonce = Nonce::from_slice(&nonce_vec);
```

Comment:
The intermediary Vec seems unnecessary here:
```suggestion
                let nonce = Nonce::from_slice(&nonce_bytes);
```

Reply:
I had rather a large amount of trouble getting the `aes_gcm` code to work, which is why I made the nonce an explicit `Vec`.  I think it's helpful for didactic purposes.